### ISSUE TITLE
PaginatedDataTable2State.dispose() needs to detach from its PaginatorController.

### ISSUE THREAD

**Issue by devrelm** [maxim-saplin/data_table_2#373]
PaginatedDataTable2State.dispose() does not call `controller.detach()` in its `dispose()` method:

https://github.com/maxim-saplin/data_table_2/blob/879f9baa25751dd928d2cc8511d95d519cd551bb/lib/src/paginated_data_table_2.dart#L580-L583

Right now, a consumer that passes a PaginatedController into a PaginatedDataTable2 is unable to use `PaginatorController.isAttached` to  tell whether they can safely call any of the methods on the controller.
